### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vasyl-Poremchuk**

### CODE REVIEW

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Review by FranchukS — CHANGES_REQUESTED** [review]
You were confused a little with the second task item.
An update/delete car buttons had to be added on the car detail page, and not in the car list table

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Review by FranchukS — APPROVED** [review]
Nice!

File: taxi/urls.py [hunk 1]
```diff
@@ -16,7 +22,25 @@
...
+    ),
+    path(
+        "manufacturers/<int:pk>/delete/",
+        ManufacturerDeleteView.as_view(),
+        name="manufacturer-delete",
+    ),
     path("cars/", CarListView.as_view(), name="car-list"),
+    path("cars/create/", CarCreateView.as_view(), name="car-create"),
+    path("cars/<int:pk>/update/", CarUpdateView.as_view(), name="car-update"),
+    path("cars/<int:pk>", CarDeleteView.as_view(), name="car-delete"),
```
Comment (Nattalli) [inline taxi/urls.py:43]: fix the path

File: templates/taxi/manufacturer_confirm_delete.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+{% block content %}
+  <h1>Delete Manufacturer</h1>
+  <p>Are you sure you want to delete the manufacturer {{ manufacturer }}?</p>
+  <p><i>All cars with this manufacture will be also deleted!</i></p>
+
+  <form action="" method="post">
+    {% csrf_token %}
+    <input type="submit" value="Yes, delete" class="btn btn-danger">
+  </form>
```
Comment (Nattalli) [inline templates/taxi/manufacturer_confirm_delete.html:11]: Can you add a "No" or "Cancel" button? This is not a mistake, but it looks nicer, more complete

File: templates/taxi/car_confirm_delete.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <h1>Delete Car</h1>
+  <p>Are you sure you want to delete the car {{ car }}?</p>
+
+  <form action="" method="post">
+    {% csrf_token %}
+    <input type="submit" value="Yes, delete" class="btn btn-danger">
+  </form>
```
Comment (Nattalli) [inline templates/taxi/car_confirm_delete.html:10]: Can you add a "No" or "Cancel" button? This is not a mistake, but it looks nicer, more complete

File: templates/taxi/car_list.html [hunk 4]
```diff
@@ -1,16 +1,36 @@
 {% extends "base.html" %}
 
 {% block content %}
-  <h1>Car list</h1>
+  <h1>
+    Car list
+    <a style="float: right" href="{% url 'taxi:car-create' %}">Add</a>
```
Comment (Nattalli) [inline templates/taxi/car_list.html:6]: Can you make this text more beautiful? In my opinion, it does not look good

**Review by Nattalli — CHANGES_REQUESTED** [review]

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Review by Nattalli — APPROVED** [review]

**Pull request closed without merging**